### ISSUE TITLE
Incorrect compiler error "Can't convert Task<T?> to Task<T>"

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#40433]
T is a class, the file where the error is reported does not have `#nullable enable` and the reported diagnostic is an error, not a warning.

**Version Used**: 

16.5.0 Preview 2.0  [29615.19.master]

**Steps to Reproduce**:

1. Checkout dotnet/roslyn @ https://github.com/dotnet/roslyn/commit/dab406578cb5dc9507796add51e849391d6e59f6
2. build

**Expected Behavior**:

No build errors.

**Actual Behavior**:

```
##[error]src\VisualStudio\RemoteHostClientMock\Remote\RemoteHostClientFactory.cs(30,20): error CS0029: Cannot implicitly convert type 'System.Threading.Tasks.Task<Microsoft.VisualStudio.LanguageServices.Remote.ServiceHubRemoteHostClient?>' to 'System.Threading.Tasks.Task<Microsoft.CodeAnalysis.Remote.RemoteHostClient>'
```

See build error:
https://dev.azure.com/dnceng/public/_build/results?buildId=459591&view=logs&j=6a7d83c1-6850-57a3-ff09-22192451ed19&t=4ea3a14e-48a8-5786-ef24-cc42a01f29bc&l=387

Reproduces locally in the IDE as well.

**Comment by tmat** [comment]
Never mind, the type is a subtype. The nullability was not relevant.
